moveit_ros/planning_interface/planning_scene_interface/src/planning_scene_interface.cpp

```diff
@@ -207,29 +209,45 @@ class PlanningSceneInterface::PlanningSceneInterfaceImpl
     moveit_msgs::PlanningScene planning_scene;
     planning_scene.world.collision_objects = collision_objects;
     planning_scene.is_diff = true;
-    planning_scene_diff_publisher_.publish(planning_scene);
+    applyPlanningScene(planning_scene);
   }
 
   void removeCollisionObjects(const std::vector<std::string> &object_ids) const
   {
     moveit_msgs::PlanningScene planning_scene;
     moveit_msgs::CollisionObject object;
-    for(std::size_t i=0; i < object_ids.size(); ++i)
+    for (std::size_t i=0; i < object_ids.size(); ++i)
     {
       object.id = object_ids[i];
       object.operation = object.REMOVE;
       planning_scene.world.collision_objects.push_back(object);
     }
     planning_scene.is_diff = true;
-    planning_scene_diff_publisher_.publish(planning_scene);
+    applyPlanningScene(planning_scene);
   }
 
 private:
 
   ros::NodeHandle node_handle_;
   ros::ServiceClient planning_scene_service_;
+  mutable ros::ServiceClient apply_planning_scene_service_;
```

Comment:
@davetcoleman The methods `addCollisionObjects() const` and `removeCollisionObjects() const` are both const .. but `ros::ServiceClient::call()` is not const. To use mutable was the simplest in order to not modify [planning_scene_interface/planning_scene_interface.h#L86-L89](https://github.com/Xamla/moveit/blob/d2511fb495168e0a2cafa3b989d9ad45798ddc6d/moveit_ros/planning_interface/planning_scene_interface/include/moveit/planning_scene_interface/planning_scene_interface.h#L86-L89).